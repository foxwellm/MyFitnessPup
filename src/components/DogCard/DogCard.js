import React from 'react'
import { Link } from 'react-router-dom'
import nopic from '../../assets/images/nopic.png'
let shortID = require('short-id');

export const DogCard = (props) => {

  const { photo, name, distance, id } = props
  const dogName = !name || !name.match(' ') ? name : name.split(' ')[0]
  return (
    <Link className='DogCard-container' to={`/dog/${id}`} key={shortID.generate()}>
      <div className='info-container'>
        <div className='info-inner-container'>
          <div className='dog-name'>{dogName}</div>
          <div className='search-img-container'>
            <img className='dog-img' alt='dog from shelter' src={photo === null ? nopic : photo}></img>
          </div>
          <div className='dog-distance'>{distance} away</div>
        </div>
      </div>
    </Link>
  )
}

export default DogCard