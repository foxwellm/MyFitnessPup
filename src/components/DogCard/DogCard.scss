.DogCard-container {
  width: 300px;
  height: 220px;
  // background-image: url('../../../public/images/siberian-husky.jpg')
  // background-size: 300px 200px;
  // background:  no-repeat center center;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  &:hover {
    background-color: blue;
  }
}

.dog-card-header {
  // height: 20px;
  margin: 0;
}

.dog-img {
  width: 100px;
  height: 200px;
  background-size: 100px 200px;
  // border-radius: 50%;
}
// .card-img {
//   box-sizing: border-box;
//    width: 100px;
//      height: 100px;
// }