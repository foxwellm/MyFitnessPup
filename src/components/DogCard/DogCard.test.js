import React from 'react'
import { shallow } from 'enzyme'
import DogCard from './DogCard'

describe('DogCard', () => {
  let wrapper
  let mockPhoto
  let mockName
  let mockAge
  let mockDistance

  beforeEach(() => {
    mockPhoto = 'photoUrl'
    mockName= 'Butch'
    mockAge= 'Young'
    mockDistance = '17.0 mi'
    wrapper = shallow(<DogCard 
        photo={mockPhoto}
        name={mockName}
        age={mockAge}
        distance={mockDistance}
      />)
  })
  it('should have match the correct snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have match the correct snapshot when the dog has multiple names', () => {
    mockName = 'Butch Casidy'
    wrapper = shallow(<DogCard
      photo={mockPhoto}
      name={mockName}
      age={mockAge}
      distance={mockDistance}
    />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should have match the correct snapshot when the dog has no name', () => {
    mockName = ''
    wrapper = shallow(<DogCard
      photo={mockPhoto}
      name={mockName}
      age={mockAge}
      distance={mockDistance}
    />)
    expect(wrapper).toMatchSnapshot()
  })
})