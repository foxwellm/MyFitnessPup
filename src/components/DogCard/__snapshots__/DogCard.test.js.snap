// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogCard should have match the correct snapshot 1`] = `
<Link
  className="DogCard-container"
  key="6"
  replace={false}
  to="/dog/undefined"
>
  <div
    className="dog-name"
  >
    Butch
  </div>
  <div
    className="dog-age"
  >
    Age: 
    Young
  </div>
  <div
    className="dog-distance"
  >
    Location: 
    17.0 mi
     away
  </div>
  <div
    className="search-img-container"
  >
    <img
      alt="dog from shelter"
      className="dog-img"
      src="photoUrl"
    />
  </div>
</Link>
`;

exports[`DogCard should have match the correct snapshot when the dog has multiple names 1`] = `
<Link
  className="DogCard-container"
  key="6"
  replace={false}
  to="/dog/undefined"
>
  <div
    className="dog-name"
  >
    Butch
  </div>
  <div
    className="dog-age"
  >
    Age: 
    Young
  </div>
  <div
    className="dog-distance"
  >
    Location: 
    17.0 mi
     away
  </div>
  <div
    className="search-img-container"
  >
    <img
      alt="dog from shelter"
      className="dog-img"
      src="photoUrl"
    />
  </div>
</Link>
`;

exports[`DogCard should have match the correct snapshot when the dog has no name 1`] = `
<Link
  className="DogCard-container"
  key="6"
  replace={false}
  to="/dog/undefined"
>
  <div
    className="dog-name"
  />
  <div
    className="dog-age"
  >
    Age: 
    Young
  </div>
  <div
    className="dog-distance"
  >
    Location: 
    17.0 mi
     away
  </div>
  <div
    className="search-img-container"
  >
    <img
      alt="dog from shelter"
      className="dog-img"
      src="photoUrl"
    />
  </div>
</Link>
`;
